mcp/gae_locals.tf

```diff
@@ -14,6 +14,9 @@ locals {
   gae                     = try(yamldecode(local.user_gae_config_yml), {})
   gae_components          = lookup(local.gae, "components", {})
   gae_components_specs    = lookup(local.gae_components, "specs", {})
+  // If environment variables exist, use that, otherwise use traffic file.
+  //`yamldecode`var.gcp_ae_traffic` is to make tha condition fail if not met, as there is no exit function in terraform
+  gae_traffic_config   = try(var.gcp_ae_traffic != null? var.gcp_ae_traffic : yamldecode(var.gcp_ae_traffic) , yamldecode(file(var.gcp_ae_traffic_yml)), {})
```

Comment:
Space. If pick on these particularly because a `?` at the end of a keyword can sometime mean 'can be null'